### ISSUE TITLE
Computer freezes as soon as training starts

### ISSUE THREAD

**Issue by SebastianKlein** [tflearn/tflearn#241]
Hi,
I'm currently have 2 hidden layers, one convolutional and one fully connected, with 200 neurons each. The input are 640x120x1 images with a batch size of 8. But as soon as the training starts, the whole computer freezes. I have waited about 30 min but nothing happens and I have to do a power reset (no kernel log is saved).

The server is running ubuntu has about 32GB ram and is using about 28Gb before the crash. Tensorflow is running in cpu mode. How can I fix this? Shouldn't tensorflow catch OOM errors?

Thanks!

**Comment by aymericdamien** [comment]
How large are your data? Maybe you are trying to load too much data in memory. You can try to use 'image_preloader' http://tflearn.org/data_utils/#image-preloader, to load your data from disk.

When you say it freezes, can you check CPU usage, to know if it really freeze or if it is just computing... Also try a smaller batch size (such as 1 just for test).

**Comment by SebastianKlein** [comment]
Thanks for your answer. My dataset has 7400 images(90% test and 10% validation). I can't do anything when it freezes. But until it freezes each CPU is running at 90% - 97%. I've tested different amounts of neurons and as soon as as the ram is about to hit it's limit the who system freezes.
I've tested the model on my pc and there I get an OOM error, on the server however not :/

I'm currently planing to reduce the image size and I'll try a batch size of 1.

**Comment by jid1** [comment]
Don't know if this has anything to do with this issue but hopefully could be useful to some people. 

Problem: 
PC freezing following CPU training required a hard reboot. Even though I tried multiple configurations, my model consisted of a few LSTM layers, adam optimizer, a BatchNormalization layer and softmax.

Solution:
I updated the motherboard BIOS "ASUS Z87-K" and until now it seems to be working fine.

Sorry for the long comment, but this should help people searching...

Extra info:
My PC was completely freezing, running on CPU. My CPU utilisation was 100% and my CPU working above the 3.5GHz at about 3.64GHz. After checking temperature (ok), forcing CPUs to work slower the issue persisted.

I installed docker assuming it could help but I got exactly the same.

The freeze tended to happen after approx. 1736 epochs and even if I run in 4 training batches (repeatedly calling fit) it would freeze at about the same number of epochs (regardless of time between runs).

System:
Windows 10 Pro (latest update)
32GB RAM
Intel i7-4770K @3.5GHz
**ASUS Z87-K**
